### PR TITLE
Add instructions to run in docker

### PR DESCRIPTION
**Pull request by frostming**
I don't know which container registry you prefer, I choose [GitHub packages](https://github.com/features/packages) in the PR.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -13,21 +13,21 @@
...
+- [Providers](#providers)
+  - [Builtin Providers](#builtin-providers)
+    - [Vagrant Provider](#vagrant-provider)
+    - [Footloose Provider](#footloose-provider)
+    - [DigitalOcean Provider](#digitalocean-provider)
+    - [Linode Provider](#linode-provider)
+  - [Write Your Own Providers](#write-your-own-providers)
+  - [Publish Your Own Providers](#publish-your-own-providers)
+- [FAQ](#faq)
+- [I Want to Know More!](#i-want-to-know-more)
```
Comment (frostming) [inline readme.md:30]: I have no preference on these change, they are made by my auto-formatter :P

Comment (laixintao) [reply readme.md:30]: I also prefer `-`, `*` was generated by my vim markdown toc plugin.

File: readme.md [hunk 2]
```diff
@@ -104,7 +106,23 @@ systemd/[supervisord](http://supervisord.org/) or put it into a docker.
...
+...
+# Run the docker container
+docker run -v `pwd`/lobbyboy_data/:/app/dev_datadir -p "12200:12200" -d ghcr.io/laixintao/lobbyboy
+```
+
+The lobbyboy server should be active on 12200 port and you can connect to it with
+
+```
+ssh Gustave@127.0.0.1 -p 12200 -i lobbyboy_data/.ssh/id_rsa
+```
```
Comment (laixintao) [inline readme.md:125]: I don't think users can run successfully follow this.

there are 2 pairs:
- lobbyboy server authenticate ssh user: using authorized_keys in config: https://github.com/laixintao/lobbyboy/blob/ecab1eea2c744834665fdfe2f6318e3d2f92b627/lobbyboy/conf/lobbyboy_config.toml#L18 , so the key pair you generated by `ssh-keygen -f lobbyboy_data/.ssh/id_rsa` won't be used.
- ssh user verify lobbyboy server: lobbyboy's key will be generated when server starts, only if not exist. here: https://github.com/laixintao/lobbyboy/blob/ecab1eea2c744834665fdfe2f6318e3d2f92b627/lobbyboy/socket_handle.py#L127

So for saving users' trouble, I suggest that we commit the ssh keys in git, so users can just start a docker, then ssh into it (always).

```
mkdir lobbyboy_data/.ssh
ssh-keygen -f lobbyboy_data/.ssh/id_rsa
````

those won't be needed.

---

I think we need to print a warning message when the server start with the default trusted pub key, and refused to listen on 0.0.0.0 for safety consideration.

Comment (frostming) [reply readme.md:125]: > there are 2 pairs:

I don't see why there are TWO pairs, from my testing I must add the pubkey into `authorized_key` and use `ssh -i private_key` to connect. The existing one in the example config seems not used.

Comment (laixintao) [reply readme.md:125]: hmm sorry, I just checked the code. seems @luxiaba 's refactor changed this behavior.

in short:

the `data_dir/.ssh/id_rsa` is not for you to use this key to login, instead, it is the server's key, which lobbyboy will show you (the pubkey finger print actually), for you to trust it (there is no middle man).

Comment (laixintao) [reply readme.md:125]: this is too confusing, let me check it back.

Comment (laixintao) [reply readme.md:125]: I have submitted a PR to you:

https://github.com/frostming/lobbyboy/pull/1

**Comment by frostming** [comment]
> So for saving users' trouble, I suggest that we commit the ssh keys in git, so users can just start a docker, then ssh into it (always).

If I understand correctly the `.ssh/id_rsa` must be exposed to the client for connecting, right? So user must prepare three parts: 1) a docker container, 2) a config file 3) the private key?

**Comment by laixintao** [comment]
> If I understand correctly the .ssh/id_rsa must be exposed to the client for connecting, right?

Yes

> So user must prepare three parts: 1) a docker container, 2) a config file 3) the private key?

Yes, but by default, users can use the default config (and we should keep it runnable always) and `dev_datadir/.ssh/id_rsa` to ssh in. so (2) and (3) can be provider by us by default. (maybe including `-v dev_datadir:dev_datadir` in docker start command?

**Comment by frostming** [comment]
> (maybe including -v dev_datadir:dev_datadir in docker start command?
Yes, but since `dev_datadir` can't be auto-generated before starting the server, this directory must be prepared by the users.

It would be better if there are no manual steps except for a config file.

**Comment by laixintao** [comment]
> Or alternatively, use a default password for connection instead of key pairs.

I prefer this, commit default username/password and key pairs to repo, in `dev_datadir`. https://github.com/lobbyboy-ssh/lobbyboy/tree/main/dev_datadir . Since the main goal of the readme is only a quick taste, not a secure, complete setup.

So one can start it with a single docker command without configuring anything.

But we must warn user to change default credentials before deploying them to production.

**Comment by frostming** [comment]
Oh, sorry i had misunderstood in the last comment, yes, we can mount the `dev_datadir` and generate a config file for the users. Users only need to create an empty directory and start the docker container! I'll update the instructions later.

**Comment by laixintao** [comment]
> But the warning message cannot be printed on the client-side, I didn't find a better way for doing this.

Yes, but I don't think this will be a problem.

Since the warning should be printed to stderr when lobbyboy starts (there is no "client" to print, when the server just started, right?)

**Comment by laixintao** [comment]
looks good to me now! no private keys builtin, very well

🍨

**Review by laixintao — APPROVED** [review]

**Pull request merged**